content/blog/next-level-iac-breakpoint-debugging/index.md

```diff
@@ -0,0 +1,317 @@
+
+At this point a debugging client (such as VS Code) will control the flow of the program, moving forward step-by-step, one line of code at a time. But how does it know which line of code it's executing? When you run in debug mode, the binary code is instrumented with metadata about the line of code it represents. The debugger will tell the runtime, via the debugging protocol, "run until you reach line 109 in file `data.ts`, then pause and wait for a signal from me before continuing".
+
+While paused, the protocol also allows you to inspect the values of any variables that are in scope at that moment. A more sophisticated debugger will provide a [REPL][repl] environment where you can execute ad-hoc code in the scope of the line where you are paused. This can be very powerful to inspect the values of various objects and to see in real-time what the effects of executing variations of code would be. After a debugging session you can take what you've learned and edit the code accordingly.
+
+## Example: Attaching a debugger to a TypeScript Pulumi program in VS Code
+
+For this example, we will set up a new Pulumi project from the command-line and then set up VS Code to run that program in a debugger.
+
+### Step 1: Create a Pulumi project from a template
+
+For this example, we will create a new Pulumi project using the [TypeScript AWS static website template][typescript-aws-static-website-template]. Make sure you've set up Pulumi, logged into Pulumi Cloud, and configured your AWS credentials. Follow the prompts in the pulumi CLI to configure your project.
+
+```shell
+$ mkdir good-morning && cd good-morning 
+$ pulumi new static-website-aws-typescript
+
+This command will walk you through creating a new Pulumi project.
+
+Enter a value or leave blank to accept the (default), and press <ENTER>.
+Press ^C at any time to quit.
+
+project name (good-morning):
+project description (A TypeScript program to deploy a static website on AWS):
+Created project 'good-morning
+
+Please enter your desired stack name.
+To create a stack in an organization, use the format <org-name>/<stack-name> (e.g. `acmecorp/dev`).
+stack name (dev):
+Created stack 'dev'
+
+aws:region: The AWS region to deploy into (us-west-2):
+errorDocument: The file to use for error pages (error.html):
+indexDocument: The file to use for top-level pages (index.html):
+path: The path to the folder containing the website (./www):
+Saved config
+
+Installing dependencies...
+
+
+added 294 packages, and audited 295 packages in 7s
+
+43 packages are looking for funding
+  run `npm fund` for details
+
+found 0 vulnerabilities
+Finished installing dependencies
+
+Your new project is ready to go! ✨
+
+To perform an initial deployment, run `pulumi up`
+```
```

Comment:
Do we need this code block? I suspect we just tell them to create a new project uusing the static website -aws-typescript template. we couuld also just provide the commands but not all the shell outputs? @interurban what is the guidance for these kinds of things?